cmake/modules/SetUpFreeBSD.cmake

```diff
@@ -0,0 +1,149 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+set(ROOT_PLATFORM freebsd)
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES x86_64 OR CMAKE_SYSTEM_PROCESSOR MATCHES amd64)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL Intel)
+    set(ROOT_ARCHITECTURE freebsdx8664icc)
+  else()
```

Comment:
Is there really an Intel compiler for FreeBSD? I'd throw out anything that is not needed / tested on FreeBSD, probably only leave `freebsdadm64` as `ROOT_ARCHITECTURE`

Reply:
I got a request from the port maintainer (@eamjensen) to keep (and add/fix some of the FBSD specific naming) the cpu architectures in the SetupFreeBSD. Maybe refs to the old freebsd archs can be removed (freebsd4,freebsd5,freebsd7 in config/root-config.in and a bunch of options in etc/Makefile.arch) - instead we could glob/wildcard as they all seem to to the same thing for now.
I'll remove the intel thing, it makes no sense indeed.

Reply:
I cleaned up a bit, all in new commits.